plugins/Compressor/CompressorControlDialog.h

```diff
@@ -0,0 +1,234 @@
+/*
+ * CompressorControlDialog.h
+ *
+ * Copyright (c) 2020 Lost Robot <r94231@gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef COMPRESSOR_CONTROL_DIALOG_H
+#define COMPRESSOR_CONTROL_DIALOG_H
+
+#include "EffectControlDialog.h"
+#include <QLabel>
+#include <QBasicTimer>
+#include "Knob.h"
+#include "../Eq/EqFader.h"
+#include "MainWindow.h"
+#include "GuiApplication.h"
+#include "PixmapButton.h"
+#include <QMouseEvent>
+#include <QTime>
+
+
+constexpr int COMP_MILLI_PER_PIXEL = 6;
+constexpr int MIN_COMP_SCREEN_X = 800;
+constexpr int MIN_COMP_SCREEN_Y = 360;
+constexpr int MAX_COMP_SCREEN_X = 1920;
+constexpr int MAX_COMP_SCREEN_Y = 1080;
+constexpr int COMP_SCREEN_X = 1000;
+constexpr int COMP_SCREEN_Y = 700;
+constexpr int KNEE_SCREEN_X = COMP_SCREEN_Y;
+constexpr int KNEE_SCREEN_Y = COMP_SCREEN_Y;
+constexpr int COMP_KNEE_LINES = 20;
+constexpr int COMP_BOX_X = 720;
+constexpr int COMP_BOX_Y = 280;
+
+constexpr float COMP_NOISE_FLOOR = 0.000001;// -120 dbFs
+
+// -2.2 seems to be the best value for this.
+// I've seen some compressors choose -1, however.
+constexpr float COMP_LOG = -2.2;
```

Comment:
FYI, some articles say there's no industry standard for the definition of attack and release values.

Reply:
And I think it should be in `Compressor.h`(probably with `COMP_NOISE_FLOOR`).